ios/HeapSettings.bundle/generate_settings_common.rb

```diff
@@ -2,23 +2,34 @@
 
 require 'fileutils'
 
-def perform_substitution(dir, app_id_dev, app_id_prod)
+def perform_substitution(dir, json)
+  dev = json.fetch('dev', {})
+  prod = json.fetch('prod', {})
+  default = json.fetch('default', {})
+
   template = File.read File.join(dir, 'HeapSettings.plist.template')
   output_path = File.join dir, 'HeapSettings.plist'
 
   settings = template
-    .gsub('__HEAP_APP_ID_DEV__', app_id_dev || '')
-    .gsub('__HEAP_APP_ID_PROD__', app_id_prod || '')
+    .gsub('__HEAP_APP_ID_DEV__', get_app_id_for_config(dev, default) || '')
```

Comment:
would it make sense to return the default empty string value as part of `get_app_id_for_config`?

Reply:
Hmm that would be reasonable, though I think I slightly prefer the current pattern of "`nil` if no app ID, then set to an empty string in the actual plist"